plugin/src/main/java/io/jenkins/plugins/casc/impl/DefaultConfiguratorRegistry.java

```diff
@@ -134,7 +134,13 @@ private Configurator internalLookup(Type type) {
         }
 
         if (Descriptor.class.isAssignableFrom(clazz)) {
-            return new DescriptorConfigurator((Descriptor) jenkins.getExtensionList(clazz).get(0));
+            ExtensionList extensions = jenkins.getExtensionList(clazz);
+            if (extensions.size() >= 1) {
+                return new DescriptorConfigurator((Descriptor) extensions.get(0));
+            } else {
+                logger.warning("Configuration-as-Code can't handle type "+ type);
+                return null;
```

Comment:
You should not return early, think it would be better to do something like this

```java
            ExtensionList extensions = jenkins.getExtensionList(clazz);
            if (!extensions.isEmpty()) {
                return new DescriptorConfigurator((Descriptor) extensions.get(0));
            }
```

Reply:
Great suggestion @casz ! Will push the change now 👍